src/protocols/permissions.ts

```diff
@@ -274,4 +354,35 @@ export class PermissionsProtocol {
   private static isRecordPermissionScope(scope: PermissionScope): scope is RecordsPermissionScope {
     return scope.interface === 'Records';
   }
+
+
+  /**
+   * Validates scope.
+   */
+  private static validateScope(scope: PermissionScope): void {
+    if (!this.isRecordPermissionScope(scope)) {
+      return;
+    }
+    // else we are dealing with a RecordsPermissionScope
+
+    // `schema` scopes may not have protocol-related fields
+    if (scope.schema !== undefined) {
+      if (scope.protocol !== undefined || scope.contextId !== undefined || scope.protocolPath) {
+        throw new DwnError(
+          DwnErrorCode.PermissionsProtocolValidateScopeSchemaProhibitedProperties,
+          'Permission grants that have `schema` present cannot also have protocol-related properties present'
+        );
+      }
+    }
+
+    if (scope.protocol !== undefined) {
+      // `contextId` and `protocolPath` are mutually exclusive
+      if (scope.contextId !== undefined && scope.protocolPath !== undefined) {
+        throw new DwnError(
+          DwnErrorCode.PermissionsProtocolValidateScopeContextIdProhibitedProperties,
+          'Permission grants cannot have both `contextId` and `protocolPath` present'
+        );
+      }
+    }
```

Comment:
For my own understanding, do we disallow these types of things to avoid inconsistent values?

Ie a `contextId` that does not match the `protocol` provided, or `protocol` being redundant with `protocolPath` provided?

Reply:
Good question man, shows you are paying amazing attention. It is more to do with aligning encryption key scope. Currently encryption key support are derived based on either schema, protocol paths, or context. There is no point in giving out a permission with greater scope if the private key you have access to can only have a less scope.

Reply:
Ah ok that makes sense! Didn't think of that for encryption, thanks!